<!DOCTYPE html>
<html>
  <!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
  
  <title>SPARK - Qian&#39;s blog</title>
  <link rel="icon" href="https://i.imgur.com/SrZt9VU.jpg" type="image/x-icon">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0">
  
  <meta name="keywords" content=tools,principle,spatial transcriptomics>
  
  
  
    <link rel="alternate" href="/atom.xml " title="Qian&#39;s blog" type="application/atom+xml">
  

  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 6.3.0"></head>

  <body>
    <div class="container">
      <header class="header">
  <div class="blog-title">
    <a href="/" class="logo">Qian&#39;s blog</a>
    <div class="subtitle"></div>
  </div>
  <nav class="navbar">
    <ul class="menu">
      
        <li class="menu-item">
          <a href="/" class="menu-item-link">Home</a>
        </li>
      
        <li class="menu-item">
          <a href="/about" class="menu-item-link">About</a>
        </li>
      
        <li class="menu-item">
          <a href="/atom.xml" class="menu-item-link">RSS</a>
        </li>
      
    </ul>
  </nav>
</header>

<article class="post">
  <div class="post-title">
    <h1 class="article-title">SPARK</h1>
  </div>

  <!-- æ–‡ç« ç›®å½• -->
  <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#what-is-spark"><span class="toc-text">What is SPARK?</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#sparks-principle"><span class="toc-text">SPARK&#39;s principle</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#how-to-calculate-the-p-value"><span class="toc-text">How to calculate the p-value</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#reference"><span class="toc-text">Reference</span></a></li></ol>

  <div class="post-content">
    <h1 id="what-is-spark">What is SPARK?</h1>
<p>SPARK directly models spatial count data through generalized linear
spatial models. It relies on recently developed statistical formulas for
hypothesis testing, effectively controlling type I errors and yielding
high statistical power.</p>
<p>This literature thinks there are three disadvantages in the two
existing approaches to identifying spatial variable genes(SVGs).</p>
<p>Firstly, SpatialDE and Trendsceek transform count data into
normalized data, which may fail to account for the mean-variance
relationship in raw counts.</p>
<p>Secondly, SpatialDE relies on asymptotic normality and minimal
<em>P</em>-value-combination rules for constructing hypothesis tests
which may cause many true SVGs ignored.</p>
<p>Thirdly, like Trendsceek, its module is not efficient which causes
amounts of calculation time.</p>
<h1 id="sparks-principle">SPARK's principle</h1>
<p>Now, I would like to choose the SPARK principle.</p>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/SPARK_1.png" /></p>
<p>It builds upon a generalized linear spatial model (GLSM) with a
variety of spatial kernels to accommodate count data generated from
smFISH or sequencing-based spatial transcriptomics studies. With a newly
developed penalized quasi-likelihood (PQL) algorithm, SPARK is scalable
to analyzing tens of thousands of genes across tens of thousands spatial
locations. Importantly, SPARK relies on a mixture of Ï‡2 distributions to
serve as the exact test statistics distribution and takes advantage of a
recently developed Cauchy combination rule to combine information across
multiple spatial kernels for calibrated <em>P</em> value calculation. As
a result, SPARK properly controls for type I error at the
transcriptome-wide level and is more powerful for identifying SE genes
than existing approaches.</p>
<p><span class="math inline">\(y_i\left(s_i\right) \sim
\operatorname{Poi}\left(N_i\left(s_i\right)
\lambda_i\left(s_i\right)\right), i=1,2, \cdots, n\)</span></p>
<p><span class="math inline">\(y_i\left(s_i\right)\)</span> as the gene
expression measurement in terms of counts for the <em>i</em>â€™th
sample.</p>
<p><span class="math inline">\(N_i(s_i)\)</span> as the normalization
factor for i'th sample. Here, it set <span
class="math inline">\(N_i(s_i)\)</span> as the summation of the total
number of counts across all genes for the sample as the main interest is
in analyzing the relative gene expression level.</p>
<p><span class="math inline">\(\log
\left(\lambda_i\left(s_i\right)\right)=\boldsymbol{x}_i\left(s_i\right)^{\boldsymbol{T}}
\boldsymbol{\beta}+b_i\left(s_i\right)+\epsilon_i\)</span></p>
<p>where <span class="math inline">\(\lambda_i\left(s_i\right)\)</span>
is an unknown Poisson rate parameter that represents the underlying gene
expression level for the <em>i'</em>th sample. In the spatial setting,
<span class="math inline">\(\lambda_i\left(s_i\right)\)</span> can also
be viewed as the unobserved spatial random process occurring at location
<span class="math inline">\(ğ’”_ğ‘–\)</span>. We model the log scale of the
latent variable <span
class="math inline">\(\lambda_i\left(s_i\right)\)</span> as a linear
combination of three terms.</p>
<p><span class="math inline">\(\log
\left(\lambda_i\left(s_i\right)\right)=\boldsymbol{x}_i\left(s_i\right)^T
\boldsymbol{\beta}+b_i\left(s_i\right)+\epsilon_i\)</span></p>
<p><span class="math inline">\(x_i\left(s_i\right)\)</span> is
<em>k</em>-vector of covariates that include a scalar of one for the
intercept and <em>k</em>-1 observed explanatory variables for the i'th
sample.</p>
<p><strong>ğœ·</strong> is a <em>k</em>-vector of coefficients that
include an intercept representing the mean log-expression of the gene
across spatial locations together with <em>k</em>-1 coefficients for the
corresponding covariates; âˆˆğ‘– is the residual error that is independently
and identically distributed from ğ‘(0,<span
class="math inline">\(ğœ^2\)</span>) with variance <span
class="math inline">\(ğœ^2\)</span>; and <span
class="math inline">\(ğ‘_ğ‘–(ğ’”_ğ‘–)\)</span> is a zero-mean, stationary
Gaussian process modeling the spatial correlation pattern among spatial
locations</p>
<p><span
class="math inline">\(\boldsymbol{b}(s)=\left(b_1\left(s_1\right),
b_2\left(s_2\right), \cdots, b_n\left(s_n\right)\right)^T \sim
\operatorname{MVN}\left(0, \tau_1 \boldsymbol{K}(s)\right)\)</span></p>
<p>The covariance <span class="math inline">\(K(s)\)</span> is a kernel
function of the spatial locations
<strong>ğ’”</strong>=(<strong>ğ’”</strong>1,â€¦,<strong>ğ’”</strong>ğ‘›)ğ‘‡, with
<em>ij</em>â€™th element being ğ‘²(<span
class="math inline">\(ğ’”_ğ‘–\)</span>,<span
class="math inline">\(ğ’”_ğ‘—\)</span>); ğœ1 is a scaling factor of the
covariance kernel; and MVN denotes a multivariate normal
distribution.</p>
<p>This is the defined GLSM model.</p>
<p>testing whether a gene shows spatial expression pattern can be
translated into testing the null hypothesis <span
class="math inline">\(ğ‘¯_0\)</span>:ğœ1=0. The statistical power of such a
hypothesis test will inevitably depend on how the spatial kernel
function <span class="math inline">\(K(s)\)</span> matches the true
underlying spatial pattern displayed by the gene of interest.</p>
<p>The true underlying spatial pattern for any gene is unfortunately
unknown and may vary across genes. To ensure robust identification of SE
genes across various spatial patterns, SPARK considers using a total of
ten different spatial kernels, including five periodic kernels with
different periodicity parameters and five Gaussian kernels with
different smoothness parameters.</p>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/SPARK_2.png" /></p>
<h1 id="how-to-calculate-the-p-value">How to calculate the p-value</h1>
<p>First, SPARK tests the null hypothesis using the ten kernels one at a
time. By one designed algorithm, to get the p-value. Afterward, it
combines these ten <em>p</em>-values through the recently developed
Cauchy <em>p</em>-value combination rule. To apply the Cauchy
combination rule, we convert each of the ten <em>p</em>-values into a
Cauchy statistic, aggregate the ten Cauchy statistics through summation,
and convert the summation back to a single <em>p</em>-value based on the
standard Cauchy distribution, because the Cauchy rule takes advantage of
the fact that combination of Cauchy random variables also follows a
Cauchy distribution regardless whether these random variables are
correlated or not</p>
<h1 id="reference">Reference</h1>
<ul>
<li>[1] <a
target="_blank" rel="noopener" href="https://www.nature.com/articles/s41592-019-0701-7">Statistical
analysis of spatial expression patterns for spatially resolved
transcriptomic studies</a></li>
</ul>

  </div>
  <div class="post-meta">
    <span class="post-time" style="color:#646060;font-size:13px">2021-03-14</span>
  </div>

  <div class="post-footer">
    
      <ul class="post-tag-list" itemprop="keywords"><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/principle/" rel="tag">principle</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/spatial-transcriptomics/" rel="tag">spatial transcriptomics</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/tools/" rel="tag">tools</a></li></ul>
    

    <a href="#top" class="top">Back to Top</a>
  </div>
</article>

<!-- è¯„è®ºæ’ä»¶ -->

<footer>
  2023
  <span class="author">
    Qian Li
  </span>
</footer>


    </div>
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ["$","$"], ["\\(","\\)"] ],
            skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code'],
            processEscapes: true
        }
    });
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax();
        for (var i = 0; i < all.length; ++i)
            all[i].SourceElement().parentNode.className += ' has-jax';
    });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>

</body>
</html>